### PR TITLE
GAWB-2833: clone synopsis

### PR DESCRIPTION
**Pull request by davidangb**
* correct the validators for POST ,/methods to be more accurate
* add tests for the validators on POST /nethods
* return a bunch route rejections and route exceptions as json, not text

I tested with a local UI - I can both clone a method with an 80 char synopsis, and when I clone with a longer synopsis, I get a real error display in the UI instead of "render failed".

hint: to get the UI to send a synopsis > 80 chars, use the DOM inspector and set the maxlength on that input field to a larger number.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aednichols** [comment]
~I am still getting a "Render Failed" when cloning with a 120-character synopsis~

Never mind, I think some part of my local setup was incorrect and/or misbehaving. Works now.

**Comment by aednichols** [comment]
(I'm trying to figure out what's up but my local setup is having auth problems.)

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala [hunk 1]
```diff
@@ -129,18 +129,144 @@ class AgoraMethodsSpec extends ApiServiceSpec with FlatSpecLike {
...
+    Post(ApiUtil.Methods.withLeadingVersion, testAgoraEntity.copy(synopsis = testSynopsis)) ~>
+      wrapWithRejectionHandler {
+        methodsService.postRoute
+      } ~> check {
+      assert(!handled)
+      assert(rejections.contains(ValidationRejection("Synopsis must be less than 80 chars",None)))
+    }
+  }
+
+  "Agora" should "return a 400 bad request when posting with a synopsis of 80 characters" in {
```
Comment (aednichols) [inline src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala]: `400 bad request` does not seem right here (+ a few other places)

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala [hunk 2]
```diff
@@ -129,18 +129,144 @@ class AgoraMethodsSpec extends ApiServiceSpec with FlatSpecLike {
...
+      wrapWithRejectionHandler {
+        methodsService.postRoute
+      } ~> check {
+      assert(!handled)
+      assert(rejections.contains(ValidationRejection("Documentation must be less than 10kb",None)))
     }
   }
 
-  "Agora" should "store 10kb of github markdown as method documentation and return it without alteration" in {
+  "Agora" should "reject 10kb of github markdown as method documentation" in {
```
Comment (aednichols) [inline src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala:269]: Good catch, this was quite misleading. I'm guessing this used to be a positive test, and then we changed it to negative?

**Review by aednichols — APPROVED** [review]
Looks good, just some naming (OK to merge without re-review after fixing)

- Tested with synopsis = 80 characters (works)
- Tested with synopsis > 80 characters (appropriate error message)

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala (hunk 2 above)
Comment (davidangb) [reply src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala:269]: I'm guessing the same thing; I don't have history on this.

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala (hunk 1 above)
Comment (davidangb) [reply src/test/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraMethodsSpec.scala]: good catch, thank you!

**Pull request merged**